### PR TITLE
Use http api forms

### PR DESCRIPTION
**Pull request by jkarni**
Replaces #582 . I'm doing this because having branches with numbers as names confuses read-the-docs, and I want to delete that one.

### CODE REVIEW

File: servant-client/servant-client.cabal [hunk 1]
```diff
@@ -41,7 +41,7 @@ library
     , base64-bytestring     >= 1.0.0.1  && < 1.1
     , bytestring            >= 0.10     && < 0.11
     , exceptions            >= 0.8      && < 0.9
-    , http-api-data         >= 0.1      && < 0.3
+    , http-api-data         >= 0.3      && < 0.4
```
Comment (soenkehahn) [inline servant-client/servant-client.cabal]: Yes. I think this is better than supporting older versions with CPP.

File: servant-server/src/Servant/Server/Internal.hs [hunk 2]
```diff
@@ -22,43 +22,40 @@ module Servant.Server.Internal
...
                                               QueryParam, QueryParams, Raw,
-                                              RemoteHost, ReqBody, Vault,
+                                              ReflectMethod (reflectMethod),
+                                              RemoteHost, ReqBody, Vault, Verb,
                                               WithNamedContext)
 import           Servant.API.ContentTypes    (AcceptHeader (..),
                                               AllCTRender (..),
-                                              AllCTUnrender (..),
-                                              AllMime,
+                                              AllCTUnrender (..), AllMime,
```
Comment (soenkehahn) [inline servant-server/src/Servant/Server/Internal.hs]: Are all these formatting changes intended? I usually try hard to avoid these. (Was there anything else changed, apart from formatting?)

Comment (jkarni) [inline servant-server/src/Servant/Server/Internal.hs]: Removed the `Web.HttpApiData.Internal` import (in favor of `Web.HttpApiData`). Yeah, this is the result of `stylish-haskell`. Not quite sure how to deal with them... Should I undo them?

Comment (soenkehahn) [inline servant-server/src/Servant/Server/Internal.hs]: Well, personally I try very hard not to make formatting changes in normal PRs. Just to keep the noise level down and to avoid (possibly tool-supported) edit wars. Otherwise I would also change formatting all the time. So from my perspective: yes, please revert.

**Comment by jkarni** [comment]
@soenkehahn fixed the imports.

File: servant-client/test/Servant/ClientSpec.hs [hunk 3]
```diff
@@ -30,7 +30,7 @@ module Servant.ClientSpec where
 import           Control.Applicative        ((<$>))
 #endif
 import           Control.Arrow              (left)
-import           Control.Concurrent         (forkIO, killThread, ThreadId)
+import           Control.Concurrent         (ThreadId, forkIO, killThread)
```
Comment (soenkehahn) [inline servant-client/test/Servant/ClientSpec.hs]: There's still some formatting changes to imports like this one that I would remove. In some other places, too.

Comment (jkarni) [inline servant-client/test/Servant/ClientSpec.hs]: What's the underlying principle, though? That it be a separate PR? Or a separate commit?

Comment (soenkehahn) [inline servant-client/test/Servant/ClientSpec.hs]: I wouldn't make these changes at all. If you're not touching code, don't change formatting.

Comment (jkarni) [inline servant-client/test/Servant/ClientSpec.hs]: But it's a pain to be consistent about formatting without using automated tools, and automated tools will touch parts that are inconsistent. My opinion is that we should aim to follow some stylish-haskell configuration, and *then* running the tool wouldn't change anything besides what you touched.  But if we still haven't reached that stage, we'll have issues like this.

Comment (soenkehahn) [inline servant-client/test/Servant/ClientSpec.hs]: My guess would be that it'd be hard to come to a consensus, which style and tool to use for code-formatting. For example, I personally use https://github.com/zalora/normalize-imports. And I also don't think it's worth the effort, at least not at this point.

I *do* think that we shouldn't have formatting changes scattered around in unrelated PRs and commits like this. In my opinion, if you touch a piece of code, please, go nuts and reformat at will. If you don't, then don't.

What I usually do is to run my code-formatting tools during development and they might change formatting where I don't want them to. Then I commit changes selectively with `git add -p`, while not staging the formatting changes in parts of the code that I didn't really touch. Afterwards I run `git checkout -p`, to revert the unwanted changes the tools introduced.

I think another good workflow that I heard about is this: You configure your editor to allow you to format only regions of the code. Like the selected code block, or the enclosing expression.

**Comment by jkarni** [comment]
@soenkehahn I fixed the imports - properly this time, I think.

File: servant/src/Servant/API/ContentTypes.hs [hunk 4]
```diff
@@ -290,12 +287,12 @@ instance OVERLAPPABLE_
          ToJSON a => MimeRender JSON a where
     mimeRender _ = encode
 
--- | @encodeFormUrlEncoded . toFormUrlEncoded@
+-- | @urlEncodeAsForm"
```
Comment (soenkehahn) [inline servant/src/Servant/API/ContentTypes.hs]: What's with the weird quote?

**Comment by soenkehahn** [comment]
@jkarni: Apart from the one comment, LGTM. (Sorry about being so pedantic about the import formatting.)

**Comment by jkarni** [comment]
> (Sorry about being so pedantic about the import formatting.) 

I was just thinking how great it is that we have you to be pedantic! This would otherwise turn into a zoo ; )

(Still not sure I agree with the general policy about import formatting, though!)
